paddle/fluid/framework/ir/mkldnn/quant_dequant_mkldnn_pass.cc

```diff
@@ -96,7 +96,50 @@ void QuantDequantMkldnnPass::CollectInfoFromFake(
   }
 }
 
-void QuantDequantMkldnnPass::CollectInputScalesFromFake(
+void QuantDequantMkldnnPass::CollectWeightScalesInfoFromONNXFormatDequantize(
+    ir::Graph* graph,
+    Scope* scope,
+    std::unordered_map<std::string, std::vector<float>>* weight_thresholds,
+    std::unordered_map<std::string, std::vector<float>>* var_quant_scales,
+    bool* onnx_format_quantize_model) const {
+  VLOG(3) << "gather weight_thresholds from onnx format dequantized ops";
+  for (auto* op_node :
+       ir::TopologyVarientSort(*graph, static_cast<ir::SortKind>(0))) {
+    if (!op_node->IsOp()) continue;
+
+    if (op_node->Name() == "dequantize_linear") {
+      auto* op_desc = op_node->Op();
+      auto x_var_name = op_desc->Input("X")[0];
+      auto* weight_var = scope->FindVar(x_var_name);
+      if (!weight_var) {
+        auto out_var_name = op_desc->Output("Y")[0];
+        if (var_quant_scales->count(x_var_name) &&
+            !var_quant_scales->count(out_var_name)) {
+          std::vector<float> scale_v = var_quant_scales->at(x_var_name);
+          var_quant_scales->insert(std::make_pair(out_var_name, scale_v));
+        }
+        continue;
+      }
+      *onnx_format_quantize_model = true;
+      auto scale_name = op_desc->Input("Scale")[0];
+      auto* var = scope->FindVar(scale_name);
+      PADDLE_ENFORCE_NOT_NULL(
+          var,
+          platform::errors::NotFound(
+              "The Scales variable [%s] of dequantize op is not found.", var));
+
+      auto* scale_tensor = var->GetMutable<LoDTensor>();
+      auto* scale_data = scale_tensor->data<float>();
+      std::vector<float> thresholds{};
+      for (int i = 0; i < scale_tensor->numel(); i++) {
+        thresholds.push_back(scale_data[i]);
+      }
```

Comment:
I like to avoid push_back if possible because vector capacity grows as more elements are pushed into it. It can generate unnecessary memory copying. Probably it doesn't matter if we have usually 1 or few values but if we can avoid it anyway I think it's worth it.

If you want, one of the ways you could write this is as below:
```suggestion
      std::vector<float> thresholds(scale_data, scale_data + scale_tensor->numel());
```

Reply:
Thanks ! Done